docs/docsite/rst/playbook_guide/playbook_pathing.rst

```diff
@@ -21,25 +21,30 @@ Task paths
 
-Resolving local relative paths
-------------------------------
+How plugins resolve local relative paths
+----------------------------------------
 
 When you specify a relative path for a local file, Ansible will try to find that file first in the current task's role, then in other roles that included or depend on the current role, then relative to the file in which the task is defined, and finally relative to the current play. It will take the first matching file that it finds. This way, if multiple files with the same file name exist, Ansible will find the file that is closest to the current task and that is most likely to be file you wanted.
 
-Specifically, Ansible tries to find the file
+Specifically, Ansible tries to find the file in the following order:
 
 1. In the current role.
 
    1. In its appropriate subdirectory—"files", "vars", "templates" or "tasks", depending on the kind of file Ansible is searching for.
    2. Directly in its directory.
    
-2. Like 1, in the parent role that called into this current role with `include_role`, `import_role`, or with a role dependency. If the parent role has its own parent role, Ansible will repeat this step with that role.
+2. Like 1, in the parent role that called into this current role with ``include_role``, ``import_role``, or with a role dependency. If the parent role has its own parent role, Ansible will repeat this step with that role.
 3. Like 1, in the current task file's directory.
 4. Like 1, in the current play file's directory.
 
-Ansible does not search the current working directory. (The directory you're in when you execute Ansible.) Also, Ansible will only search within a role if you actually included it with an `include_role` or `import_role` task or a dependency. If you instead use `include`, `include_task` or `import_task` to include just the tasks from a specific file but not the full role, Ansible will not search that role in steps 1 and 2.
+Ansible does not search for local files in the current working directory, in other words, the directory from which you execute Ansible.
 
-When you execute Ansible, the variable `ansible_search_path` will contain the paths searched, in the order they were searched in but without listing their subdirectories. If you run Ansible in verbosity level 5 by passing the `-vvvvv` argument, Ansible will report each directory as it searches, except when it searches for a tasks file.
+.. note:: When resolving local relative paths for tasks files, Ansible uses the context of the role that includes tasks with an ``include_role`` or ``import_role`` statement. If you import the tasks with ``include_task``, or ``import_task`` statements, Ansible uses the context of the importing file.
+
+Troubleshooting search paths
```

Comment:
when using include/import_tasks while targetting a file in a role, you will not load that role into the current context, so that role will not be added to the search paths. This does not alter the current context of the include so any other roles/plays alraedy in there will still be there.

Reply:
@bcoca, thank you for the comment; I think I understand now.

Rephrasing your comment, would this formulation of the note be more suitable?
"When you use include_tasks or import_tasks to include tasks from another role, Ansible does not add that role to the current context, and as a result, Ansible does not add it to the search paths. To include such tasks, use a relative path from the directory where the including statement resides."

Reply:
> I think this should go below the last note. Otherwise, that note is kind of dangling at the bottom of the page and cut off from the main section.

You mean the Troubleshooting subsection, right?

In that case, I agree; I think the last note should go directly beneath the line "Ansible does not search for local files in the current working directory, in other words, the directory from which you execute Ansible." as it adds more context to what current working dir means in different situations.

But would it be okay if there were two notes stacked right on top of each other?